### PR TITLE
feat: support to list datatype

### PR DESCRIPTION
**Pull request by WenyXu**
related #14

### CODE REVIEW

**Comment by Jefffrey** [comment]
I will take a look later today (or over weekend) :+1:

File: src/reader/schema.rs [hunk 1]
```diff
@@ -278,10 +278,11 @@ pub fn create_schema(types: &[Type], root_column: usize) -> Result<Arc<TypeDescr
             );
 
             let td = Arc::new(TypeDescription::new(ARRAY.clone(), root_column));
-            let fields = &root.field_names;
-            for (idx, column) in sub_types.iter().enumerate() {
+
+            for (_, column) in sub_types.iter().enumerate() {
```
Comment (Jefffrey) [inline src/reader/schema.rs:282]: If not using index then no need to enumerate?

File: src/arrow_reader/column/list.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        for _ in 0..chunk {
+            match self.present.next() {
+                Some(present) => {
+                    if present {
+                        match self.lengths.next().unwrap() {
+                            Ok(len) => {
+                                let _ = self
+                                    .inner
+                                    .append_value(&mut root_builder.values().builder, len as usize);
+                            }
```
Comment (Jefffrey) [inline src/arrow_reader/column/list.rs:34]: Is there any way to reduce this nesting?

File: src/arrow_reader.rs [hunk 3]
```diff
@@ -634,6 +677,9 @@ impl Decoder {
                     append_struct_null(idx, filed, builder, &iter.decoders[idx])?;
                 }
             }
+            &Decoder::List(_) => {
+                unreachable!("unreachable append null list")
```
Comment (Jefffrey) [inline src/arrow_reader.rs:681]: Why is this unreachable?

File: src/reader/schema.rs (hunk 1 above)
Comment (WenyXu) [reply src/reader/schema.rs:282]: Good point!

File: src/arrow_reader.rs (hunk 3 above)
Comment (WenyXu) [reply src/arrow_reader.rs:681]: I'm not sure here. I will add some code here and more tests in the following PRs.

**Comment by WenyXu** [comment]
PTAL @Jefffrey

**Review by Jefffrey — APPROVED** [review]
Looks good but I'm still struggling with grasping the overall structure of the code for the whole crate, so I might go through and suggest refactor PRs afterwards if that's fine

**Pull request merged**